File: ruby/lib/onvo/dashboard/sessions.rb
# frozen_string_literal: true

require_relative './dashboard_resource'

# Dashboard session endpoints
class DashboardSessions < DashboardResource
  def list
    base_get("/dashboards/#{@dashboard_id}/sessions")
  end

  # TODO: check if rename to delete_all
  def delete
    base_delete("/dashboards/#{@dashboard_id}/sessions")
  end

  def upsert(user_id, parameters = {})
    session_data = base_post(
      "/dashboards/#{@dashboard_id}/sessions",
      user: user_id,
      parameters: parameters
    )
    session_data.merge({ 'url': "#{@endpoint}#{session_data['url']}" })
  end
end

File: js/lib/sessions.js
import OnvoBase from "./base";
export class OnvoSessions extends OnvoBase {
    // Dashboard Session endpoints
    list(filters) {
        return this.fetchBase("/api/sessions?dashboard=" + filters.dashboard);
    }
    delete(id) {
        return this.fetchBase("/api/sessions/" + id, "DELETE");
    }
    revokeAll(filters) {
        return this.fetchBase("/api/sessions?dashboard=" + filters.dashboard, "DELETE");
    }
    async upsert({ user, dashboard, parameters, }) {
        let data = await this.fetchBase("/api/sessions", "POST", {
            user: user,
            parameters: parameters,
            dashboard: dashboard,
        });
        return { ...data, url: this.endpoint + data.url };
    }
}
